### PR TITLE
[docs] updates docs to use ember-cli-addon-docs

### LINKED ISSUE #78: Use ember-cli-addon-docs for built site
The current site does not allow us to search for information and is cumbersome. When you click into a section, it does not maintain a left rail of all the other options. What are the thoughts on using the default https://github.com/ember-learn/ember-cli-addon-docs to get a better user experience with the docs?

### PR DESCRIPTION
**Pull request by gabrielcsapo**
Looking at the open issues I wanted to add additional docs, but those should probably be done after we merge this, as this is a large change as it is.

# What does this look like?

Screen Shot 2020-03-26 at 8 14 51 PM
Screen Shot 2020-03-26 at 8 14 49 PM

# How do I test this locally

Pulling down this branch and running `ember s` will allow you to navigate to `http://localhost:4200`

fixes #78

### CODE REVIEW

**Comment by gabrielcsapo** [comment]
Thoughts on this? @efx @dgeb @villander

**Comment by villander** [comment]
Thanks for pushing this @gabrielcsapo, an amazing job! 
I'll review it this weekend. 

> Looking at the open issues I wanted to add additional docs, but those should probably be done after we merge this, as this is a large change as it is.

What additional docs, do you like to add? Is there something align with #55 ?

**Comment by gabrielcsapo** [comment]
Yeah I think adding additional docs and potentially cleaning up the doc hierarchy can be done after this is merged.

**Comment by villander** [comment]
Awesome, thanks for it. I'll review it as soon as possible to unblock you ; ) 

If you have some time, let's talk on Discord ; )

File: package.json
Comment (villander) [inline package.json]: can you add `ember-engines` to make it more readable in searches

File: package.json
Comment (villander) [inline package.json:39]: how can we manage to deploy here? As I know this codebase already deploys on master when CI is green. 

If we going to change  it, maybe we need rethink our infrastructure on Heroku

File: package.json
Comment (villander) [inline package.json]: I'd like to remove `.github.io` and continue using `ember-engines.com`

**Review by villander — CHANGES_REQUESTED** [review]
Thank you! 

I've some notes.

File: tests/dummy/app/index.html
Comment (villander) [inline tests/dummy/app/index.html]: we need keep this favicon and also find a place to use our brand image - https://avatars3.githubusercontent.com/u/23064483?s=200&v=4

File: package.json
Comment (gabrielcsapo) [reply package.json:39]: This will currently deploy to gh-pages branch, do we deploy from that?

Comment (villander) [reply package.json:39]: Currently, we are doing deploy automatically to Heroku. I prefer GitHub pages, but we need to change it and make sure that it will work fine without some unavailability. And make sure that GitHub pages will have the current site URL/domain.

cc: @dgeb

File: tests/dummy/app/index.html
Comment (gabrielcsapo) [reply tests/dummy/app/index.html]: The favicon is currently in the project as public/images/favicon, I added this back.

**Comment by villander** [comment]
Waiting for @dgeb support to change deployment to Github pages using the same domain.

**Comment by gabrielcsapo** [comment]
@villander looking into this, this will not only support deployment to GitHub Pages but also heroku. Heroku seems to just be running ember serve based on the revert of the prember code.

**Comment by gabrielcsapo** [comment]
it seems like I commented on an issue about moving to GitHub last year https://github.com/ember-engines/ember-engines.com/issues/42#issuecomment-530151420 🤦‍♂

**Review by villander — APPROVED** [review]

**Pull request merged**
